### DISCUSSION TITLE
Default field values in header files

### DISCUSSION THREAD

**Discussion by cx1111** [wfdb/wfdb-spec#7] [General]
According to https://www.physionet.org/physiotools/wag/header-5.htm, a header with a missing sampling frequency field is assumed to have a fs of 250Hz. Similarly, a header with a missing base_time has its value set to 00:00:00.

Should these values be part of the file specification, or just a behavior of the WFDB software package? 

I'd definitely like to remove these values from the spec, but we may not be able to, especially if there are known databases that depend on this.

**Comment by cx1111** [comment]
Also other fields:
- units: default = mV. 
- ADC resolution: 12 or 10 or 8, depending on format. This one is rather strange.

For all of these cases, does the library assume these defaults, or does the specification indicate that that's what the missing value means?

**Comment by cx1111** [comment]
We have come to consensus. We will be retrospectively naming a v10 header file spec that uses the WFDB software package v10.X's behavior, for all these fields, including defaults.

A future header file format can require the fields to be explicit, among other things.

**Comment by bemoody** [comment]
> a header with a missing base_time has its value set to 00:00:00

This is incorrect, or at least misleading.

If the record doesn't have a base time specified, then the WFDB library won't convert time values to time-of-day (instead, mstimstr always returns elapsed time.)  If you try to convert *from* time of day to elapsed time, then yeah, the library will assume the record starts at midnight; this seems like a bug.

*Most* records don't have a base time specified, and that accurately reflects the reality that we don't know the precise recording time - and applications shouldn't treat this the same way as a record that starts at midnight.

**Reply by cx1111** [reply]
Good to know. Will specify in the header spec that there is no default.

**Reply by cx1111** [reply]
Also, could you please tell me the date that the first WFDB v10.X was released?

**Comment by bemoody** [comment]
WFDB 10.0.0 was released January 15, 2000, apparently.

You can see the tags here: https://github.com/bemoody/wfdb/tags

**Reply by cx1111** [reply]
It looks like the GitHub tag was created in 2014: https://github.com/bemoody/wfdb/releases/tag/10.0.0

Just checking that it was originally released in 2000?

**Comment by bemoody** [comment]
Silly me, I thought GitHub displayed the commit author dates.  Yes, I
created the tags in 2014 when I imported the history into Git.

Look at the dates in the git log, and look at
https://archive.physionet.org/physiotools/archives/ if you don't
believe me. ;)
